### DISCUSSION TITLE
Using curl to upload a file automatically selects the available nodes

### DISCUSSION THREAD

**Discussion by xiyou123321** [curl/curl#12460] [Q&A]
When using curl to upload files, you can configure Dr Processing because my server has multiple network ports. For example, if I have two network ports that can be connected, each upload file will be iterated in order. If the network port is available and can be uploaded, the next network port will not be used. When the preferred node goes offline, it automatically switches to the next node. Verification is required for each upload to ensure that the preferred network port can be put into use after recovery. I am using curl to use ftp/ftps/sftp currently can not find this effect of the protocol, please support the available protocol and can give a description! Thank you very much

**Comment by xiyou123321** [comment]
This one, for example, supports multiple users     CURLOPT_FTP_ALTERNATIVE_TO_USER

**Comment by xiyou123321** [comment]
This is the version of curl I used        curl --version
curl 7.29.0 (x86_64-redhat-linux-gnu) libcurl/7.29.0 NSS/3.53.1 zlib/1.2.7 libidn/1.28 libssh2/1.8.0
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp scp sftp smtp smtps telnet tftp
Features: AsynchDNS GSS-Negotiate IDN IPv6 Largefile NTLM NTLM_WB SSL libz unix-sockets

**Comment by dfandrich** [comment]
I'm not sure what you're asking.  curl will connect to whatever URL you give
it. If you give it a specific port, it will use that port. If the application
wants to use a different port, just set a different port in the URL. The
application is free to do that if a request to the first port fails. This is
true for all curl versions (and note that ver. 7.29.0 is over a decade old).

**Reply by xiyou123321** [reply]
I hope that the connection failure of the first port can automatically switch to the second port. In the next upload, if the first port is restored, I will check again whether the first port is restored, and if the first port is restored, I will use the first port. But I have not found a suitable way to use it

**Reply by bagder** [reply]
Is the "port" you speak of a network interface? If so, curl uses the network interface that your routing table says is the suitable. You can override that by telling curl what interface to bind to. curl cannot by itself just pick one.

**Comment by dfandrich** [comment]
The application is free to implement that kind of logic itself, but it's not done
automatically by curl.
